protocol/agents.json

```diff
@@ -0,0 +1,85 @@
+{
+    "agents": [
+        {
+            "identifier": "ably-js",
+            "emitters": [
+                "ably-js"
+            ]
+        },
+        {
+            "identifier": "node",
+            "emitters": [
+                "ably-js"
+            ]
+        },
+        {
+            "identifier": "react-native",
+            "versioned": false,
+            "emitters": [
+                "ably-js"
+            ]
+        },
+        {
+            "identifier": "ably-cocoa",
+            "emitters": [
+                "ably-cocoa"
+            ]
+        },
+        {
+            "identifier": "iOS",
```

Comment:
Two of the product requirements are:

> Usage of different wrapper libraries (flutter, asset-tracking, etc)
> What versions of OSs and runtime environments are used?

So I think it would be good to add some extra classification of each agent in this list (e.g. `agent_type: "sdk|os|wrapper"`) that will be emitted along with the metric, which will simplify getting answers to the above questions with queries like:

- `SELECT * FROM agents WHERE agent_type = 'os'`
- `SELECT * FROM agents WHERE agent_type = 'wrapper'`

Reply:
I like that idea, but maybe the root `agents` value could be a map where the keys of that map are the agent type and then the values are arrays. Given that, if I interpret correctly, an agent will only ever have one type.

Reply:
Either of these options sound good to me, @lmars @cruickshankpg do either of you have any preference so I can go ahead and add this?

Reply:
From a router perspective, this would be easier as fields on each entry so it can be deserialised into a struct like:

```go
type Agent struct {
	Identifier string `json:"identifier"`
	Type       string `json:"agent_type"`
}
```

then when we do the lookup to find the `Agent` object for a given string, we can easily just reference `agent.Type` when emitting the metric.

Reply:
I've added a type property in 8896442